corehq/apps/users/views/mobile/users.py

```diff
@@ -1685,6 +1687,25 @@ def link_connectid_user(request, domain):
         return HttpResponse()
 
 
+@csrf_exempt
+@login_or_basic_ex(allow_cc_users=True)
+def connectid_messaging_key(request, domain):
+    link = get_object_or_404(ConnectIDUserLink, commcare_user=request.user, domain=request.domain)
+    key = generate_aes_key().decode("utf-8")
+    messaging_key = ConnectIDMessagingKey.objects.create(connectid_user_link=link, domain=request.domain, key=key)
+    return JsonResponse({"key": messaging_key.key})
+
+
+@csrf_exempt
+@require_POST
+@login_or_basic_ex(allow_cc_users=True)
+def update_connectid_messaging_consent(request, domain):
```

Comment:
This will come from the connectid server, not a specific user.

Reply:
Added. [b9bc775](https://github.com/dimagi/commcare-hq/pull/35212/commits/b9bc77504d169cf56e05cd6f2b35545b1d800bcc)